bundles/easy-api/src/FondOfOryx/Zed/EasyApi/Business/Model/ApiWrapper.php

```diff
@@ -0,0 +1,178 @@
+<?php
+
+namespace FondOfOryx\Zed\EasyApi\Business\Model;
+
+use FondOfOryx\Zed\EasyApi\Dependency\Client\EasyApiToGuzzleClientInterface;
+use FondOfOryx\Zed\EasyApi\EasyApiConfig;
+use Generated\Shared\Transfer\EasyApiFilterTransfer;
+use Generated\Shared\Transfer\EasyApiRequestTransfer;
+use Generated\Shared\Transfer\EasyApiResponseTransfer;
+use GuzzleHttp\Exception\RequestException;
+use Psr\Log\LoggerInterface;
+
+class ApiWrapper implements ApiWrapperInterface
+{
+    /**
+     * @var \FondOfOryx\Zed\EasyApi\Dependency\Client\EasyApiToGuzzleClientInterface
+     */
+    protected EasyApiToGuzzleClientInterface $guzzle;
+
+    /**
+     * @var \FondOfOryx\Zed\EasyApi\EasyApiConfig
+     */
+    protected EasyApiConfig $config;
+
+    /**
+     * @var \Psr\Log\LoggerInterface
+     */
+    protected LoggerInterface $logger;
+
+    /**
+     * @param \FondOfOryx\Zed\EasyApi\Dependency\Client\EasyApiToGuzzleClientInterface $guzzleClient
+     * @param \FondOfOryx\Zed\EasyApi\EasyApiConfig $config
+     * @param \Psr\Log\LoggerInterface $logger
+     */
+    public function __construct(EasyApiToGuzzleClientInterface $guzzleClient, EasyApiConfig $config, LoggerInterface $logger)
+    {
+        $this->guzzle = $guzzleClient;
+        $this->config = $config;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\EasyApiFilterTransfer $filterTransfer
+     *
+     * @return \Generated\Shared\Transfer\EasyApiResponseTransfer
+     */
+    public function findDocument(EasyApiFilterTransfer $filterTransfer): EasyApiResponseTransfer
+    {
+        $uri = sprintf('%s/%s', $this->config->getEasyApiUri(), 'api/content/search');
+
+        $response = $this->request($uri, 'post', $filterTransfer);
```

Comment:
@julianzimmermann should $this->request be changed in $this->guzzle oder i'am wrong?